### PR TITLE
Update to Kotlin 1.8.20

### PR DESCRIPTION
**Pull request by ZacSweers**

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -4,7 +4,7 @@ autoService = "1.0.1"
 autoValue = "1.10"
 dagger = "2.39.1"
 espresso = "3.5.1"
-kotlin = "1.8.0"
+kotlin = "1.8.20-Beta"
```
Comment (JoelWilcox) [inline gradle/libs.versions.toml:7]: Remember to update the kotlin matrixes in ci.yml for testing too

Comment (ZacSweers) [reply gradle/libs.versions.toml:7]: 4d0f4ba

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -19,7 +19,7 @@ jobs :
       # solution.
       fail-fast : false
       matrix :
-        kotlin-version : [ 1.8.0 ]
+        kotlin-version : [ 1.8.20-Beta ]
```
Comment (JoelWilcox) [inline .github/workflows/ci.yml:22]: There are a couple other jobs that also have this matrix as well like `test-ubuntu`

File: gradle/libs.versions.toml [hunk 3]
```diff
@@ -14,7 +14,7 @@ autoValue = "1.10"
 dagger = "2.39.1"
 espresso = "3.5.1"
 gradlePublish = "0.15.0"
-kotlin = "1.8.0"
+kotlin = "1.8.20"
```
Comment (JoelWilcox) [inline gradle/libs.versions.toml:17]: Can you also update the ext property `kotlinPreview`? It's currently set based on `1.8.0`

Comment (ZacSweers) [reply gradle/libs.versions.toml:17]: [e4a604a](https://github.com/square/anvil/pull/689/commits/e4a604a8ef3e19bdfa8ef91a06f8d043263910e8)

**Review by JoelWilcox — APPROVED** [review]

**Pull request merged**